raiden/tests/unit/transfer/mediated_transfer/test_mediation_fee.py

```diff
@@ -434,6 +430,69 @@ def test_fee_round_trip(flat_fee, prop_fee, imbalance_fee, amount, balance1, bal
     assert amount_with_margin_after_fees >= amount
 
 
+@example(flat_fee=0, prop_fee=0, imbalance_fee=1277, amount=1, balance1=33, balance2=481)
+@given(
+    integers(min_value=0, max_value=100),
+    integers(min_value=0, max_value=10_000),
+    integers(min_value=0, max_value=50_000),
+    integers(min_value=1, max_value=90_000_000_000_000_000_000),
+    integers(min_value=1, max_value=100_000_000_000_000_000_000),
+    integers(min_value=1, max_value=100_000_000_000_000_000_000),
+)
+def test_fee_add_remove_invariant(flat_fee, prop_fee, imbalance_fee, amount, balance1, balance2):
+    """ First adding and then removing fees must yield the original value """
+    # Find examples where there is a reasonable chance of succeeding
+    amount = int(min(amount, balance1 * 0.95 - 1, balance2 * 0.95 - 1))
+    assume(amount > 0)
+
+    total_balance = TokenAmount(100_000_000_000_000_000_000)
+    prop_fee_per_channel = ppm_fee_per_channel(ProportionalFeeAmount(prop_fee))
+    imbalance_fee = calculate_imbalance_fees(
+        channel_capacity=total_balance,
+        proportional_imbalance_fee=ProportionalFeeAmount(imbalance_fee),
+    )
+    fee_schedule = FeeScheduleState(
+        cap_fees=False,
+        flat=FeeAmount(flat_fee),
+        proportional=prop_fee_per_channel,
+        imbalance_penalty=imbalance_fee,
+    )
+    channel_in = factories.create(
+        NettingChannelStateProperties(
+            our_state=NettingChannelEndStateProperties(balance=total_balance - balance1),
+            partner_state=NettingChannelEndStateProperties(balance=balance1),
+            fee_schedule=fee_schedule,
+        )
+    )
+    channel_out = factories.create(
+        NettingChannelStateProperties(
+            our_state=NettingChannelEndStateProperties(balance=balance2),
+            partner_state=NettingChannelEndStateProperties(balance=total_balance - balance2),
+            fee_schedule=fee_schedule,
+        )
+    )
+
+    amount_with_fees = get_amount_with_fees(
+        amount_without_fees=amount,
+        schedule_in=channel_in.fee_schedule,
+        schedule_out=channel_out.fee_schedule,
+        receivable_amount=balance1,
+        balance_in=total_balance - balance1,
+        balance_out=balance2,
+    )
+    assume(amount_with_fees)
+    assert amount_with_fees
```

Comment:
Do we need both assume and assert?

Reply:
The assert is for mypy, which does not understand how `assume` works.